tsdb/wal/wal_test.go

```diff
@@ -204,6 +204,56 @@ func TestWALRepair_ReadingError(t *testing.T) {
 	}
 }
 
+// TestUnsequentialSegments tests whether the Unsequential wal segments are
+// deleted or not.
+func TestUnsequentialSegments(t *testing.T) {
+	dir, err := ioutil.TempDir("", "wal_unsequential")
+	testutil.Ok(t, err)
+	defer func() {
+		testutil.Ok(t, os.RemoveAll(dir))
+	}()
+
+	var (
+		logger         = testutil.NewLogger(t)
+		segmentSize    = pageSize * 3
+		recordSize     = pageSize - recordHeaderSize
+		segments       = 20
```

Comment:
`numSegments` would sound better here, I confused it with actual segments below.